weave/engine_trace.py

```diff
@@ -17,10 +17,32 @@
 import time
 import json
 import dataclasses
+from typing import Optional
 
 from . import logs
 from . import stream_data_interfaces
 
+from ddtrace import span as ddtrace_span
```

Comment:
as a practical matter, it is typically best practice to put monkey patch logic in a function that runs one and only once.

Reply:
Will move to within tracer.trace